### PR TITLE
add show query executor

### PR DESCRIPTION
**Pull request by TatsuNet**
add executor for show query

### CODE REVIEW

File: exec/show.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+}
+
+// QueryRow doesn't support in ShowQueryExecutor, returns always error.
+func (e *ShowQueryExecutor) QueryRow() (*sql.Row, error) {
+	return nil, errors.New("ShowQueryExecutor cannot invoke QueryRow()")
+}
+
+// Exec executes `SHOW TABLE` DDL for shards
+func (e *ShowQueryExecutor) Exec() (sql.Result, error) {
+	debug.Printf("show table for shards")
```
Comment (goccy) [inline exec/show.go]: Not only is this method called on `show create table`, there are other possibilities (e.g. `show status` ). 
Therefore, this API should implement switching to `show create table` routine if `Table` parameter exists .

Comment (goccy) [reply exec/show.go]: Also, I think `SHOW CREATE TABLE` expect `QueryRow` API. Please confirm !

Comment (TatsuNet) [reply exec/show.go]: Change to support `Query()` and `QueryRow()`, dosen't support `Exec()`. `Query()` and `QueryRow()` return result of any one shards.

**Comment by TatsuNet** [comment]
Fixed redundant logic.

**Comment by goccy** [comment]
LGTM !!!

**Pull request merged**